lecture/build.gradle

```
@@ -0,0 +1,289 @@
+
+            // de.tum.in.www1.artemis.web.rest
+            include 'de/tum/in/www1/artemis/web/rest/dto/**'
+            include 'de/tum/in/www1/artemis/web/rest/errors/EntityNotFoundException.*'
+            include 'de/tum/in/www1/artemis/web/rest/errors/BadRequestAlertException.*'
+            include 'de/tum/in/www1/artemis/web/rest/errors/FieldErrorVM.*'
+            include 'de/tum/in/www1/artemis/web/rest/errors/StudentsAlreadyAssignedException.*'
+            include 'de/tum/in/www1/artemis/web/rest/errors/InternalServerErrorException.*'
+            include 'de/tum/in/www1/artemis/web/rest/errors/AccessForbiddenException.*'
+            include 'de/tum/in/www1/artemis/web/rest/errors/ErrorConstants.*'
+        }
+        resources {
+            srcDir '../src/main/resources'
+            include 'config/liquibase/**'
+            include 'config/application-artemis.yml'
+            include 'config/application-gitlab.yml'
+            include 'config/application-jenkins.yml'
+        }
+    }
+    externalTest {
+        java {
+            srcDir file("../src/test")
+            include "java/de/tum/in/www1/artemis/util/RequestUtilService.*"
+        }
+    }
+    main {
+        compileClasspath += external.java
+        java {
+            srcDir external.java
+        }
+        resources {
+            srcDir external.resources
+        }
+    }
+    test {
+        compileClasspath += external.java
+        compileClasspath += externalTest.java
+
+        java {
+            srcDir external.java
+            srcDir externalTest.java
+        }
+    }
+}
+
+idea {
+    module {
+        excludeDirs += files("node_modules")
+    }
+}
+
+spotless {
+    // allows to execute the code formatting commands ./gradlew :lecture:spotlessApply and ./gradlew :lecture:spotlessCheck
+    java {
+        target project.fileTree(project.rootDir) {
+            include "lecture/**/*.java"
+            exclude "**/src/main/java/de/tum/in/www1/artemis/lecture/config/SecurityConfiguration.java", "**/node_modules/**",
+                "**/out/**", "**/repos/**", "**/build/**", "**/src/main/generated/**", "**/src/main/resources/templates/**",
+                "**/exports/**", "src/main/java/de/tum/in/www1/artemis/**"
+        }
+        importOrderFile "../artemis-spotless.importorder"
+        eclipse().configFile "../artemis-spotless-style.xml"
+
+        removeUnusedImports()
+    }
+}
+
+defaultTasks "bootRun"
+
+springBoot {
+    mainClass = "de.tum.in.www1.artemis.lecture.LectureApp"
+}
+
+bootRun {
+    jvmArgs = ["--illegal-access=warn"]
+}
+
+modernizer {
+    failOnViolations = true
+    includeTestClasses = true
+}
+
+configurations {
+    providedRuntime
+}
+
+apply from: "../gradle/dependencies.gradle"
+
+dependencies {
+    implementation "com.offbytwo.jenkins:jenkins-client:0.3.8"
+    implementation "org.gitlab4j:gitlab4j-api:4.18.0"
+    implementation files("../libs/jplag-3.0.0.jar")
```

```suggestion
    implementation "com.offbytwo.jenkins:jenkins-client:${offbytwo_jenkins_client_version}"
    implementation "org.gitlab4j:gitlab4j-api:${gitlab4j_api_version}"
    implementation files("../libs/jplag-${jplag_version}.jar")
```
Can variables be used for the version numbers as well?

---

Yes, I've missed that